batch/batch/front_end/front_end.py

```diff
@@ -579,14 +580,23 @@
                 secrets = spec.get('secrets')
                 if not secrets:
                     secrets = []
-                    spec['secrets'] = secrets
+
+                for secret in secrets:
+                    if user != 'ci':
+                        raise web.HTTPBadRequest(reason=f'unauthorized secret {(secret["namespace"], secret["name"])}')
+
+                spec['secrets'] = secrets
                 secrets.append({
                     'namespace': BATCH_PODS_NAMESPACE,
                     'name': userdata['gsa_key_secret_name'],
                     'mount_path': '/gsa-key',
                     'mount_in_copy': True
                 })
 
+                sa = spec.get('service_account')
+                if sa and user != 'ci' and not (user == 'test' and sa['name'] == 'test-batch-sa' and sa['namespace'] == BATCH_PODS_NAMESPACE):
+                    raise web.HTTPBadRequest(reason=f'unauthorized service account {(sa["namespace"], sa["name"])}')
```

Comment:
I think this would be easier to understand as:
```
sa = spec.get('service_account')
raise_if_user_lacks_permission_for_service_account(user, sa)
# ...
def raise_if_user_lacks_permission_for_service_account(user, sa):
    if sa is None:
        return
    if user == 'ci':
        return
    if user == 'test':
        if sa['name'] == 'test-batch-sa'  and sa['namespace'] == BATCH_PODS_NAMESPACE:
            return
    raise web.HTTPBadRequest(reason=
        f'unauthorized service account {(sa["namespace"], sa["name"])} for user {user}')
```
The model is: state what is permitted, everything else is considered an error. When checking for security, I want to enumerate the permitted states and make sure those seem right.

Reply:
In fact, can we declare the allowed service accounts for ci as well? Presumably there's only a couple that are allowed?